Cleaned up ItemExpireEvent

HenryLoenwind opened the pull request:
The old code had grown a bit of bitrot. Also cleaned up the event itself, getting rid of the superfluous cancelability. Otherwise the functionality is the same.

HenryLoenwind committed:
Cleaned up ItemExpireEvent

The old code had grown a bit of bitrot. Also cleaned up the event itself, getting rid of the superfluous cancelability. Otherwise the functionality is the same.

Spinoscythe commented on src/main/java/net/neoforged/neoforge/event/entity/item/ItemExpireEvent.java:
```diff
@@ -6,32 +6,33 @@
+    /**
+     * Sets the amount of extra life time (in ticks) to give this EntityItem.
+     * 
+     * @param extraLife The amount of time to be added to this entities lifespan.
```
You should use inline javadocs here AFAIK, ignore this if not.

HenryLoenwind replied:
Could you explain what you mean with "inline javadocs"?

TelepathicGrunt replied:
Not sure what to inline here either. Seems fine to me

TelepathicGrunt commented on src/main/java/net/neoforged/neoforge/event/entity/item/ItemExpireEvent.java:
```diff
@@ -6,32 +6,33 @@
+     * 
+     * @param extraLife The amount of time to be added to this entities lifespan.
+     */
     public void setExtraLife(int extraLife) {
```
Potentially silly suggestion. How about adding `addExtraLife` to make it clear to modders that they can add onto other people's extra life instead of overwriting? People might forget that is an option when doing set

HenryLoenwind replied:
Not so silly at all, I also was contemplating adding addExtraLife(). It just seemed a bit superfluous, as set(get()+x) already works fine and should be obvious.

HenryLoenwind replied:
Seeing I wasn't the only one with that idea: Done.

HenryLoenwind committed:
Merge remote-tracking branch 'neoforged/1.21.x' into itemexpirecleanup

XFactHD commented on src/main/java/net/neoforged/neoforge/event/entity/item/ItemExpireEvent.java:
```diff
@@ -6,32 +6,33 @@
- * lifespan. Canceling this event will prevent the EntityItem from being
- * flagged as dead, thus staying it's removal from the world. If canceled
- * it will add more time to the entities life equal to extraLife.
+ * lifespan. Adding extra life time will prevent the EntityItem from being
```
Since you're touching the docs anyway: please replace `EntityItem` with `{@link ItemEntity}` throughout the class

HenryLoenwind replied:
Will do...once my installation has recovered from switching branches.

HenryLoenwind replied:
Done

HenryLoenwind committed:
More javadoc, fixed javdoc, added add()

XFactHD commented on src/main/java/net/neoforged/neoforge/event/EventHooks.java:
```diff
@@ -515,10 +515,9 @@ public static PlayLevelSoundEvent.AtPosition onPlaySoundAtPosition(Level level,
     }
 
-    public static int onItemExpire(ItemEntity entity, ItemStack item) {
-        if (item.isEmpty()) return -1;
```
The empty check should probably be retained, except that it would return 0 now.

HenryLoenwind replied:
The empty check is pointless as the item cannot be empty here---the event is fired in an isEmpty()-protected block. The only way it could be empty was if someone patched the movement code to empty it.

XFactHD replied:
Whoops, completely missed that

HenryLoenwind committed:
Fix a logic bug in the original event design

Adding to lifespan isn't that good an idea in the first place, but this way at least the foot is out of the line of fire.

Spinoscythe reviewed (approved):
LGTM

Matyrobbrt commented on src/main/java/net/neoforged/neoforge/event/entity/item/ItemExpireEvent.java:
```diff
@@ -6,33 +6,59 @@
- * it will add more time to the entities life equal to extraLife.
+ * Event that is fired when an {@link ItemEntity}'s age has reached its maximum
+ * lifespan. Adding extra life time will prevent the {@link ItemEntity} from being
+ * flagged as dead, thus staying it's removal from the world.
```
```suggestion
 * flagged as dead, thus staying its removal from the world.
```

Matyrobbrt committed:
Update src/main/java/net/neoforged/neoforge/event/entity/item/ItemExpireEvent.java
